app/src/main/java/com/goryachok/forecastapp/data/LocalDataSource.kt

```diff
@@ -0,0 +1,70 @@
+package com.goryachok.forecastapp.data
+
+import android.content.Context
+import android.util.Log
+import com.google.gson.Gson
+import com.goryachok.forecastapp.base.RemoteEntity
+import com.goryachok.forecastapp.model.domain.ForecastEntity
+import com.goryachok.forecastapp.model.domain.WeatherEntity
+import javax.inject.Inject
+import kotlin.reflect.KClass
+
+class LocalDataSource @Inject constructor(context: Context) {
+
+    companion object{
+        private const val DAGGER_TAG = "DaggerDebug"
+    }
+    init {
+        Log.d(DAGGER_TAG, javaClass.name)
+    }
+
+    private val prefsName = "WEATHER_PREFERENCES"
+    private val weatherPrefs = "LOCAL_WEATHER_DATA"
+    private val forecastPrefs = "LOCAL_FORECAST_DATA"
+
+    private val prefs = context.getSharedPreferences(prefsName, Context.MODE_PRIVATE)
+
+    fun saveData(remoteEntity: RemoteEntity) {
+        when (remoteEntity) {
+            is WeatherEntity -> {
+                val str = Gson().toJson(remoteEntity, WeatherEntity::class.java)
+                prefs.edit().putString(weatherPrefs, str).apply()
+            }
+            is ForecastEntity -> {
+                val str = Gson().toJson(remoteEntity, ForecastEntity::class.java)
+                prefs.edit().putString(forecastPrefs, str).apply()
+            }
+        }
+    }
+
+    fun lastFetchTime() = if (isDataAvailable()) getData(WeatherEntity::class).date else 0
+
+//    fun getWeatherData(): WeatherEntity {
```

Comment:
Please either add "TODO" if you're planning to use commented code, or remove unused commented code.

Reply:
Removed

Reply:
Removed